### PR TITLE
core/types: fix discrepancy in receipt.EffectiveGasPrice json encoding tags

### PR DESCRIPTION
**Pull request by roberto-bayardo**
the execution api [spec](https://github.com/ethereum/execution-apis/blob/af82a989bead35e2325ecc49a9023df39c548756/src/schemas/receipt.yaml#L49) suggests receipt.EffectiveGasPrice should be required, but it's currently not tagged as required in its receipt.go schema. Further, the field is tagged as 'omitempty' in the generated code even though it's not tagged as such in the schema.

This PR leaves the field as untagged in the receipt.go schema for backwards compatibility, adds a comment clarifying it's still technically required, removes the omitempty discrepancy in the generated code, and adds/updates test cases to codify the untagged behavior of the field. 

Alternatives considered:

- tag as required in schema + generated code. Dismissed due to backwards compatilbity concerns as it could break interop w/ older servers that do not require the field.

- tag as omitempty in both schema + generated code.  I don't have a strong reason yet to exclude this alternative, though I don't have a reason to prefer it either.

### CODE REVIEW

**Comment by fjl** [comment]
I don't like adding `required` for this field because it will break decoder compatibility with older servers that do not have the field set.

**Comment by roberto-bayardo** [comment]
> I don't like adding `required` for this field because it will break decoder compatibility with older servers that do not have the field set.

Makes sense. Would leaving it as neither required nor omit empty make most sense then? That's how it was in the schema before, but the generated code still had omit empty for reasons I don't entirely understand.

**Comment by roberto-bayardo** [comment]
Hmm. I dropped the required tag from the schema, replacing with a comment.  But the regenerated gen_receipt_json.go didn't change (other than the tag disappearing).

**Comment by roberto-bayardo** [comment]
This PR now leaves the field as untagged in the receipt.go schema for backwards compatibility, but adds a comment clarifying it's still technically required, removes the omitempty discrepancy in the generated code, and adds/updates test cases to codify the untagged behavior of the field.

Updated the PR description accordingly, and also provided a list of the alternatives considered (tag as required, tag as omitempty).

File: core/types/receipt.go [hunk 1]
```diff
@@ -62,7 +62,7 @@ type Receipt struct {
 	TxHash            common.Hash    `json:"transactionHash" gencodec:"required"`
 	ContractAddress   common.Address `json:"contractAddress"`
 	GasUsed           uint64         `json:"gasUsed" gencodec:"required"`
-	EffectiveGasPrice *big.Int       `json:"effectiveGasPrice"`
+	EffectiveGasPrice *big.Int       `json:"effectiveGasPrice"` // required, but tag omitted for backwards compatibility
```
Comment (lightclient) [inline core/types/receipt.go:65]: Can you also please add this value to `receiptMarshaling` and regenerate? We'd like to use `hexutil.Big` to ultimately decode.

File: cmd/evm/t8n_test.go [hunk 2]
```diff
@@ -275,17 +275,18 @@ func TestT8n(t *testing.T) {
...
+			want, err := os.ReadFile(file)
 			if err != nil {
 				t.Fatalf("test %d: could not read expected output: %v", i, err)
 			}
 			have := tt.Output()
 			ok, err := cmpJson(have, want)
 			switch {
 			case err != nil:
-				t.Fatalf("test %d, json parsing failed: %v", i, err)
+				t.Fatalf("test %d file %v:, json parsing failed: %v", i, file, err)
```
Comment (karalabe) [inline cmd/evm/t8n_test.go:287]: test %d, file %v:  comma in the middle, no comma after colon

File: core/types/receipt_test.go [hunk 3]
```diff
@@ -82,27 +82,15 @@ var (
...
-	to6 := common.HexToAddress("0x6")
-	to7 := common.HexToAddress("0x7")
-	txs := Transactions{
+	to2     = common.HexToAddress("0x2")
+	to3     = common.HexToAddress("0x3")
+	to4     = common.HexToAddress("0x4")
+	to5     = common.HexToAddress("0x5")
+	to6     = common.HexToAddress("0x6")
+	to7     = common.HexToAddress("0x7")
+	testTxs = Transactions{
```
Comment (karalabe) [inline core/types/receipt_test.go:93]: Can we pls have this reverted?

File: core/types/receipt_test.go [hunk 4]
```diff
@@ -161,37 +149,38 @@ func TestDeriveFields(t *testing.T) {
...
-	blockTime := uint64(2)
-	blockHash := common.BytesToHash([]byte{0x03, 0x14})
-
-	// Create the corresponding receipts
-	receipts := Receipts{
+	// Create corresponding receipts for the testTxs
+	blockNumber  = big.NewInt(1)
+	blockTime    = uint64(2)
+	blockHash    = common.BytesToHash([]byte{0x03, 0x14})
+	testReceipts = Receipts{
```
Comment (karalabe) [inline core/types/receipt_test.go:156]: Can we pls have this reverted? (testReceipts vs receipts)

File: cmd/evm/t8n_test.go (hunk 2 above)
Comment (holiman) [inline cmd/evm/t8n_test.go:287]:
```suggestion
				t.Fatalf("test %d, file %v: json parsing failed: %v", i, file, err)
```

**Review by holiman — APPROVED** [review]
LGTM

**Pull request merged**